zoos/zoos-router.js

```diff
@@ -0,0 +1,112 @@
+const router = require("express").Router();
+const knex = require("knex");
+
+//-----KnexConfig-----//
+const knexConfig = {
+  client: "sqlite3",
+  useNullAsDefault: true,
+  connection: {
+    filename: "./data/lambda.sqlite3"
+  },
+  debug: true
+};
+
+const db = knex(knexConfig);
+
+//------ Get-----//
+
+router.get("/", (req, res) => {
+  db("zoos")
+    .then(zoos => {
+      res.status(200).json(zoos);
+    })
+    .catch(err => {
+      res.status(500).json(err);
+    });
+});
+
+//-----Get by ID-----//
+router.get("/:id", (req, res) => {
+  const { id } = req.params;
+
+  db("zoos")
+    .where({ id })
+    .first()
+    .then(zoo => {
+      res.status(200).json(zoo);
+    })
+    .catch(err => {
+      res.status(500).json(err);
+    });
+});
+
+//------Post------//
+
+router.post("/", (req, res) => {
+  if (!req.body.name) {
+    return res
+      .status(400)
+      .json({ message: "Please provide a name for the zoo." });
+  }
+  db("zoos")
+    .insert(req.body)
+    .then(ids => {
+      const id = ids[0];
+      db("zoos")
+        .where({ id })
+        .first()
+        .then(zoo => {
+          res.status(201).json(zoo);
+        });
+    })
+    .catch(err => {
+      res.status(500).json(err);
+    });
+});
+
+//------Delete-----//
+router.delete("/:id", (req, res) => {
+  const { id } = req.params;
+
+  db("zoos")
+    .where({ id })
+    .del()
+    .then(count => {
+      if (count > 0) {
+        res.status(204).end();
```

Comment:
```suggestion
        res.status(200).json(<ErrorMessage>);
```
Be careful using 204 status code as it will not send anything back.